### PR TITLE
sys.config changes taken from upstream GA release

### PR DESCRIPTION
**Pull request by pritamghanghas**
**[Issue](https://github.com/NebraLtd/hm-block-tracker/issues/84)**

- Link:https://github.com/NebraLtd/hm-block-tracker/issues/84
- Summary: Helium miner 2022.04.19.0 release has light miner built into it (disabled state). There are new variables in the upstream sys.config that needs to incorporated into our cdn sys.config

**How**
This includes three commits that has not been squashed to make reviewing easier
1. Adds config-state.template as exact replica of config.template and modifies the script to support two separate configs
2. makes changes to config-state.template to include new upstream config variables to support lightminers
3. changes snapshotter/sys.config to new values from upstream.

**Screenshots**

**References**

**Checklist**

- [ ] Tests added
- [ ] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: config-stage.template [hunk 1]
```diff
@@ -0,0 +1,148 @@
+%% -*- erlang -*-
```
Comment (pritamghanghas) [inline config-stage.template:1]: This is mostly a copy of config.template. see diff between commit 1 and 2 to see what all has changed

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,17 @@
+attrs==21.4.0
```
Comment (pritamghanghas) [inline requirements.txt:1]: was missing packages while running pytest. Exported 'pip freeze' again

File: snapshotter/sys.config [hunk 3]
```diff
@@ -47,9 +47,9 @@
    {fetch_latest_from_snap_source, true},
    {honor_quick_sync, true},
    {quick_sync_mode, blessed_snapshot},
-   {blessed_snapshot_block_height, 1160641},
+   {blessed_snapshot_block_height, 1318321},
    {blessed_snapshot_block_hash,
-    <<114,177,177,98,217,154,188,18,212,229,36,77,57,49,35,131,227,60,157,115,47,34,29,59,241,63,24,152,221,198,197,5>>},
+    <<34,205,162,107,155,68,156,241,133,93,191,151,136,126,186,148,108,195,13,62,106,205,175,236,235,131,189,106,249,216,149,117>>},
```
Comment (pritamghanghas) [inline snapshotter/sys.config:52]: Block height and hash are taken from upstream sys.config. They come out different if I use generate script. I am not sure about the relationship between this and generated file.

File: requirements.txt (hunk 2 above)
Comment (marvinmarnold) [reply requirements.txt:1]: Can we make a test-requirements.txt?

File: miner_config/generate_config.py [hunk 4]
```diff
@@ -54,14 +54,24 @@ def output_config_file(config, path):
...
+def is_rockpi():
+    return bool(int(os.getenv('ROCKPI', '0')))
+
+
 def main():
     init_sentry()
-    if bool(int(os.getenv('PRODUCTION', '0'))):
-        base_url = 'https://helium-snapshots.nebracdn.com'
-    else:
+    if is_testnet_fleet():
```
Comment (marvinmarnold) [inline miner_config/generate_config.py:67]: For consistency with [other code](https://github.com/NebraLtd/hm-block-tracker/blob/master/snapshotter/create_snapshot.sh#L7), to reduce the number of env vars, and to make testnet/staging default, I suggest:

```
if is_production():
  # prod logic
else:
  # staging logic
```

File: miner_config/generate_config.py [hunk 5]
```diff
@@ -54,14 +54,24 @@ def output_config_file(config, path):
...
+        template_path = 'config-stage.template'
+    else:
+        base_url = 'https://helium-snapshots.nebracdn.com'
+        template_path = 'config.template'
 
-    if bool(int(os.getenv('ROCKPI', '0'))):
+    if is_rockpi():
         i2c_bus = 'i2c-7'
         path = 'docker.config.rockpi'
     elif bool(int(os.getenv('PISCES', '0'))):
```
Comment (marvinmarnold) [inline miner_config/generate_config.py:77]: Why not refactor pisces too if you did it for rockpi?

File: snapshotter/sys.config (hunk 3 above)
Comment (marvinmarnold) [reply snapshotter/sys.config:52]: Agreed. I also believe these values are ultimately overwritten, but hard to follow every possible code path. Might as well update.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Awesome, this is exactly what I was expecting, just some small comments.

File: miner_config/generate_config.py (hunk 5 above)
Comment (pritamghanghas) [reply miner_config/generate_config.py:77]: missed it. Was in a hurry to sleep

File: miner_config/generate_config.py (hunk 4 above)
Comment (pritamghanghas) [reply miner_config/generate_config.py]: done

**Comment by pritamghanghas** [comment]
Fixed review comments.

File: miner_config/generate_config.py [hunk 6]
```diff
@@ -54,25 +54,36 @@ def output_config_file(config, path):
...
+    if is_prodcution_fleet():
         base_url = 'https://helium-snapshots.nebracdn.com'
+        template_path = 'config.template'
     else:
         base_url = 'https://helium-snapshots-stage.nebracdn.com'
+        template_path = 'config-stage.template'
 
-    if bool(int(os.getenv('ROCKPI', '0'))):
+    if is_device_type('ROCKPI'):
         i2c_bus = 'i2c-7'
```
Comment (marvinmarnold) [inline miner_config/generate_config.py:75]: In the future we can reference hardware definitions, but not important now: https://github.com/NebraLtd/hm-pyhelper/blob/b48f500b407d3c2ce8651844bb8a939746847691/hm_pyhelper/hardware_definitions.py#L20

**Review by marvinmarnold — APPROVED** [review]
